types/metro-resolver/errors.d.ts

```diff
@@ -0,0 +1,37 @@
+import { FileAndDirCandidates, FileCandidates } from './types';
+
+export class FailedToResolvePathError extends Error {
+    candidates: FileAndDirCandidates;
+}
```

Comment:
We should use `declare class` here, and also include the contructor:
```suggestion
export declare class FailedToResolvePathError extends Error {
    candidates: FileAndDirCandidates;
    constructor(candidates: FileAndDirCandidates);
}
```

Reply:
It's odd, the CI told me that the 'declare' keyword is redundant here: https://github.com/DefinitelyTyped/DefinitelyTyped/actions/runs/3418278974/jobs/5690377227#step:6:83

Reply:
Huh, TIL. Feel free to ignore those comments then 😅

Reply:
😆 still thanks for your patience in reviewing the code and giving comments, don't worry, I will roll these changes back